src/client/caching/structures/guilds/Guild.ts

```diff
@@ -324,22 +346,121 @@ export class Guild extends Structure {
+	 */
+	public getWidgetImageURL(options?: WidgetImageOptions): string {
+		const path = Routes.guildWidgetImage(this.id);
+		// TODO(VladFrangu): I think we should move this to @klasa/rest
+		const url = new URL(`https://discord.com/api${path}`);
+		if (options) for (const [key, value] of Object.entries(options)) url.searchParams.append(key, value);
+		return url.toString();
+	}
+
+	/**
+	 * Returns the guild preview.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/guild#get-guild-preview
+	 */
+	public fetchPreview(): Promise<APIGuildPreviewData> {
+		const endpoint = Routes.guildPreview(this.id);
+		return this.client.api.get(endpoint) as Promise<APIGuildPreviewData>;
+	}
+
+	/**
+	 * Edit's a guild's settings.
+	 * @since 0.0.1
+	 * @param data The settings to be applied to the guild.
+	 * @see https://discord.com/developers/docs/resources/guild#modify-guild
+	 */
 	public async edit(data: GuildEditOptions): Promise<unknown> {
-		return this.client.api.patch(Routes.guild(this.id), { data });
+		const endpoint = Routes.guild(this.id);
+		const result = await this.client.api.patch(endpoint, { data }) as APIGuildData;
+		return this.clone<this>()._patch(result);
 	}
 
+	/**
+	 * Delete the guild permanently.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/guild#delete-guild
+	 */
 	public async delete(): Promise<unknown> {
 		return this.client.api.delete(Routes.guild(this.id));
 	}
 
-	public async prune(options: GuildPruneOptions): Promise<unknown> {
+	/**
+	 * Returns a number indicating the number of members that would be removed in a prune operation.
+	 * @since 0.0.1
+	 * @param options The number of days to count prune and the included roles.
+	 * @param dry When set to `true`, a calculation of how many members would be pruned.
+	 * @see https://discord.com/developers/docs/resources/guild#get-guild-prune-count
+	 */
+	public prune(options: GuildPruneDryOptions, dry: true): Promise<number>;
+
+	/**
+	 * Begins a prune operation.
+	 * @since 0.0.1
+	 * @param options The number of days to count prune and the included roles.
+	 * @param dry When set to `false` or left undefined, this operation will kick users.
+	 * @see https://discord.com/developers/docs/resources/guild#begin-guild-prune
+	 */
+	public prune(options: GuildPruneOptions, dry?: false): Promise<number | null>
+	public async prune(options: GuildPruneDryOptions, dry?: boolean): Promise<number | null> {
+		const endpoint = Routes.guildPrune(this.id);
+		if (dry) {
+			const result = await this.client.api.get(endpoint, { query: options }) as { pruned: number };
+			return result.pruned;
+		}
+
 		// eslint-disable-next-line @typescript-eslint/camelcase
-		const data = { ...options, compute_prune_count: this.large };
-		return this.client.api.post(Routes.guildPrune(this.id), { data });
+		const result = await this.client.api.post(endpoint, { query: { compute_prune_count: this.large, ...options } }) as { pruned: number | null };
```

Comment:
Why..are you setting compute_prune_count based on this.large? It should be provided by the user, not defaulted